python/cudf/cudf/pandas/fast_slow_proxy.py

```diff
@@ -564,8 +582,12 @@ def _fsproxy_wrap(cls, value, func):
         _FinalProxy subclasses can override this classmethod if they
         need particular behaviour when wrapped up.
         """
-        proxy = object.__new__(cls)
-        proxy._fsproxy_wrapped = value
+        if np.ndarray in cls.__mro__:
+            proxy = ProxyNDarrayBase.__new__(cls, value)
+            proxy._fsproxy_wrapped = value
```

Comment:
This is not great, but I think its required if our `Proxy{Insert NumPy Object Name}Base` classes inheriit from their real NumPy parents. You get the `TypeError: object.__new__(cls) is not safe, use ProxyNDArrayBase.__new__()`.

Because we'll adding multiple proxy base classes, we could just reframe the check so that we're looking if it has a proxy base class. And then call `__new__` on that base class. That it's not so "on the nose" that there's NumPy specifics in this file.

Reply:
This PR _may_ help defining the proxyclass in a way that the MRO is happy: https://github.com/rapidsai/cudf/pull/15399

Reply:
I tried adding a metaclass with `__call__` that calls `np.ndarray.__new__` but I still got the same `TypeError`